gradle/root/coverage.gradle

```diff
@@ -1,43 +1,25 @@
-task mockitoCoverage(type: JacocoReport) {
-
-    allprojects { currentProject ->
-        // These tests do not have a `test` task.
-        if (currentProject.name in ['android']) {
-            return
```

Comment:
Assuming this was meant to exclude the current :androidTest module (because :android is a java module), this is now removed. As an effect of this the coverage might go down, because there are uncovered classes:
![image](https://github.com/mockito/mockito/assets/2906988/502df4fe-066b-4fee-b32e-033f948eb98a)

Reply:
We now do have tests for this module, but it does not report any coverage for these files. Can we update the android CI jobs to collect and upload the coverage as well?

Reply:
Ah, so :android doesn't have unit tests, but it has the instrumentation integration tests in :androidTest.

I would double-test (same source code running as unit on JVM and instrumentation) with Robolectric if that's possible for these. The reason being is that coverage is really hard to share between jobs, unless Codecov has some magic for this. Locally when I ran coverageReport, it executed all the tests, if we add :androidTest:testDebugAndroidTest as a dependency like others, then running covrageReport will require an emulator to be running. Will have a look at how Codecov is set up, hopefully it's doable with it.

Reply:
CodeCov already combines all code coverage reports of all test suites, including our subprojects. The last step of the pipeline does this: https://github.com/mockito/mockito/blob/c1eec880a27eee89b68b75a8995a94431ff3fdb3/.github/workflows/ci.yml#L73

I guess we need to add this last step to the android jobs as well, assuming we already gather code coverage?

Reply:
Yeah, that's what I'm doing right now.

This relies on having an .xml file filled up with data. That's why step 7 gathers all subprojects (code in coverage.gradle). So need a similar setup for `:androidTest`. I think I got it working. Pushing as soon as the current CI jobs complete, so I can observe the covarge in the below comment before/after.